openupgradelib/openupgrade_merge_records.py

```diff
@@ -368,7 +370,15 @@ def _adjust_merged_values_orm(env, model_name, record_ids, target_record_id,
         if not field.store or field.compute or field.related:
             continue  # don't do anything on these cases
         op = field_spec.get(field.name, False)
-        _list = all_records.mapped(field.name)
+        if field.type != 'reference':
+            _list = all_records.mapped(field.name)
+        else:
+            try:
+                _list = all_records.mapped(field.name)
+            except Exception:
+                # if two Reference fields records have distinct selection
+                # option, then it returns mixing apples and oranges TypeError
+                continue
```

Comment:
I'm thinking that you should build a list instead:
```suggestion
            _list = [x[field.name] for x in all_records]
```